erpnext/manufacturing/doctype/bom/bom.py

```diff
@@ -134,6 +137,30 @@ def autoname(self):
 
 		self.name = name
 
+	@staticmethod
+	def get_next_version_index(existing_boms: List[str]) -> int:
+		# split by "/" and "-"
+		delimiters = ["/", "-"]
+		pattern = "|".join(map(re.escape, delimiters))
+		bom_parts = [re.split(pattern, bom_name) for bom_name in existing_boms]
+
+		# filter out BOMs that do not follow the following formats:
+		# - BOM/ITEM/001
+		# - BOM/ITEM/001-1
+		# - BOM-ITEM-001
+		# - BOM-ITEM-001-1
+		valid_bom_parts = list(filter(lambda x: len(x) > 1 and x[-1], bom_parts))
+
+		# extract the current index from the BOM parts
+		if not valid_bom_parts:
+			# handle cancelled and submitted documents
+			indexes = [cint(part[2] for part in valid_bom_parts)]
```

Comment:
@ankush

Do you think there's a better way to do this? Should we even bother with this anymore and simply let the system fail with an error message?

If we do, and an item has a `-` or '/' character in its name, then this logic will fail.

I was thinking I could remove all amended documents entirely by adding it as a filter in the `get_all` above:

```python
existing_boms = frappe.get_all(
	  "BOM",
	  filters={"item": self.item, "amended_from": ["is", "not set"]},
	  pluck="name"
)
```

That should let me change this back to:

```suggestion
			indexes = [cint(part[-1] for part in valid_bom_parts)]
```

which, although still implicit, is better than the current state, I think.